src/wp-includes/canonical.php

```diff
@@ -740,6 +754,28 @@ function lowercase_octets( $matches ) {
 		$requested_url = preg_replace_callback( '|%[a-fA-F0-9][a-fA-F0-9]|', 'lowercase_octets', $requested_url );
 	}
 
+	if (
+		$redirect_obj &&
+		is_a( $redirect_obj, 'WP_Post' )
+	) {
+		$post_status_obj = get_post_status_object( get_post_status( $redirect_obj ) );
```

Comment:
Should this maybe be extracted to a function so we can early return for each conditional? I've had trouble grokking this.

Reply:
Could that be `wp_force_ugly_post_permalink`? It seems like the logic between this and that should be synced. It looks like this isn't currently handling if the post status isn't registered.

Reply:
@TimothyBJacobs I've cleaned up the logic in redirect_canonical in db69d13cd91bd976ba65f9694c136894ed73caa4. I think it's clearer what is happening now. There is some risk though that a post status that's registered as `private` and `internal` or `protected` will redirect but that's not intended behavior so going to let it pass.

Reply:
:+1: Agreed.